### PR TITLE
removed gdal for mbostock shapefile

### PR DESCRIPTION
**Pull request by nmccready**
- added ability to filter files in zip archive for unzipping. This allows for archives larger than one file.
- gdal was causing issues so removed it for pure javascript via mbostock shapefile
- added options on cli for shapefile encoding and ignore-properties
- supports multiple shapefiles in a zip archive

### CODE REVIEW

**Comment by nmccready** [comment]
If this starts being discussed I will kill the fork renaming.

**Comment by maxogden** [comment]
my 2 cents: rather than an 'either/or' for `gdal` vs `shapefile` i'd prefer a way to choose which one to use, as they both have their own advantages. but practically speaking neither me nor @substack are actively using this module at the moment so I don't think a fork is that bad. but I think the name `shp2jsonx` could be improved. maybe `shapefile2json`?

**Comment by nmccready** [comment]
I liked the name and I wanted to keep the reference of where it came from mostly via the x for extended. But I don't really care about the name. I just needed something different to put it on npm. Also it seems consuming stdin is not working in my PR and I am having a heck of a time of figuring it out.

File: index.js [hunk 1]
```diff
@@ -19,110 +49,146 @@ module.exports = function (inStream) {
...
+                                    next(null, out);
+                                    out = '';
+                                } else {
+                                    writeNextFeature();
+                                }
+                            });
+                        };
+                        if (firstTime) {
+                            firstTime = false;
+                            reader.readHeader(function() {
```
Comment (nmccready) [inline index.js]: This cb is not being called on consuming a stream only. I am wondering if there is some race condition. The files do exist in the /tmp/* .

**Comment by nmccready** [comment]
As far as making it support both gdal and shapefile that would take some effort for a small abstraction layer. I personally couldn't get gdal working locally. The specs would pass, but real archives would blow up.

**Comment by nmccready** [comment]
Appears the streaming piping of a zip is broken due to something wrong w/ node. When I updated to iojs-2.0.1 it works fine.

**Comment by nmccready** [comment]
I tested node 0.10.36 and 0.10.38 and no reading happens in shapefile. It really appears there is an issue in node itself.

**Comment by nmccready** [comment]
This code is never reached (and should be). https://github.com/mbostock/shapefile/blob/master/file.js#L11

**Pull request closed without merging**